src/Lecture1.hs

```diff
@@ -104,7 +124,13 @@ and finds a sum of the numbers inside this string.
 
 The string contains only spaces and/or numbers.
 -}
-strSum str = error "TODO"
+strSum :: String -> Integer
+strSum str = 
+  {- go one by one through string, accumulating a buffer of elements.
+     upon space, append buffer to result list. should handle floating points that way. -}
+  let nums = map read (words str) :: [Integer] {- or just use words I guess -}
```

Comment:
As you specify the top-level type signature for this function, the compiler would be able to figure out what type `read` function has in this case due to type inference, so you can skip specifying it in the body:

```suggestion
  let nums = map read (words str) {- or just use words I guess -}
```